consensus/istanbul/core/errors.go

```diff
@@ -49,8 +49,14 @@ var (
 	errInvalidPreparedCertificateDigestMismatch = errors.New("message in PREPARED certificate for different digest than proposal")
 	// errInvalidRoundChangeViewMismatch is returned when the PREPARED certificate view is greater than the round change view
 	errInvalidRoundChangeViewMismatch = errors.New("View for PREPARED certificate is greater than the view in the round change message")
+	// errInvalidRoundChangeRequestSignature is returned when a RoundChangeRequest is improperly signed
+	errInvalidRoundChangeRequestSignature = errors.New("Round Change Request improperly signed")
+	// errRoundChangeRequestAddressMismatch is returned when a RoundChangeRequest address does not match the message address
+	errRoundChangeRequestAddressMismatch = errors.New("Round Change Request address does not match message address")
 	// errInvalidPreparedCertificateInconsistentViews is returned when the PREPARED certificate view is inconsistent among it's messages
 	errInvalidPreparedCertificateInconsistentViews = errors.New("View is inconsistent among the PREPARED certificate messages")
+	// errRoundChangeProposalHashMismatch is returned when the proposal hash of a round change prepared certificate does not match the proposal's hash.
+	errRoundChangeProposalHashMismatch = errors.New("Proposal hash mismatch in round change message")
```

Comment:
Rather than adding these errors with static messages I would suggest instead just creating an error at the point where you are using them. This would allow for much richer error descriptions that are able to provide contextual information such as what was the proposal hash and what was the mismatching hash and who sent the mismatched hash.

Reply:
I don't like much this setup as well, but wanted to be consistent with other consensus error types

Reply:
I think it's ok to diverge here, its not like this package is particularly clean or consistent.

Using hardcoded strings like this makes troubleshooting harder, especially when you consider that we don't run validators so the best we can do is ask for the logs after a failure, so we really need to try and cram as much useful information into those logs as possible.